### PR TITLE
Warn deprecated usage of parameter names in function types

### PR DESCRIPTION
**Pull request by federicobond**
Refs #2428

### CODE REVIEW

File: libsolidity/analysis/SyntaxChecker.cpp [hunk 1]
```diff
@@ -148,3 +148,15 @@ bool SyntaxChecker::visit(PlaceholderStatement const&)
 	return true;
 }
 
+bool SyntaxChecker::visit(FunctionTypeName const& _node)
+{
+	for (auto decl : _node.parameterTypeList()->parameters())
```
Comment (chriseth) [inline libsolidity/analysis/SyntaxChecker.cpp]: Please use `auto const& decl: _node...` (`const&` plus space)

**Review by chriseth — CHANGES_REQUESTED** [review]
I wonder whether we could just turn that into an error right away. On the other hand, we should do the 0.5.0 release quite soon anyway...

**Comment by axic** [comment]
I think we're better off making it a warning as we were following this policy for a while.

File: libsolidity/analysis/SyntaxChecker.cpp (hunk 1 above)
Comment (federicobond) [reply libsolidity/analysis/SyntaxChecker.cpp]: Fixed.

**Comment by federicobond** [comment]
Good, then this is good to merge as it is.

**Pull request merged**